File: frontend/src/api/skill.ts
import { request } from './index';

export interface Skill {
  id: number;
  name: string;
  description: string;
  icon: string;
  category: string;
  prompt: string;
  tools: string;
  installed: boolean;
  markdownContent?: string;
  scriptPath?: string;
  scriptContent?: string;
  version?: string;
  author?: string;
}

export interface SkillFormData {
  name: string;
  description: string;
  icon: string;
  category: string;
  prompt: string;
  tools: string;
  markdownContent: string;
  scriptContent: string;
  version: string;
  author: string;
}

export interface ToolInfo {
  name: string;
  description: string;
  category: string;
}

export const skillApi = {
  getAllSkills: (category?: string) => {
    const url = category ? `/skills?category=${encodeURIComponent(category)}` : '/skills';
    return request<Skill[]>(url);
  },

  getCategories: () => {
    return request<string[]>('/skills/categories');
  },

  getAvailableTools: () => {
    return request<ToolInfo[]>('/skills/tools');
  },

  getInstalledSkills: () => {
    return request<Skill[]>('/skills/installed');
  },

  getSkillById: (id: number) => {
    return request<Skill>(`/skills/${id}`);
  },

  createSkill: (data: SkillFormData) => {
    return request<number>('/skills', { method: 'POST', body: data });
  },

  updateSkill: (id: number, data: SkillFormData) => {
    return request<void>(`/skills/${id}`, { method: 'PUT', body: data });
  },

  deleteSkill: (id: number) => {
    return request<void>(`/skills/${id}`, { method: 'DELETE' });
  },

  installSkill: (id: number) => {
    return request<void>(`/skills/${id}/install`, { method: 'POST' });
  },

  uninstallSkill: (id: number) => {
    return request<void>(`/skills/${id}/uninstall`, { method: 'POST' });
  },
};

File: src/main/java/com/cmbchina/orgperformance/entity/Skill.java
package com.cmbchina.orgperformance.entity;

import lombok.Data;
import java.time.LocalDateTime;

@Data
public class Skill {
    private Long id;
    private String name;
    private String description;
    private String icon;
    private String category;
    private String prompt;
    private String tools;
    private Integer isBuiltIn;
    private Integer isActive;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    
    // 技能详细说明(Markdown格式)
    private String markdownContent;
    // 脚本文件路径
    private String scriptPath;
    // 脚本内容
    private String scriptContent;
    // 版本号
    private String version;
    // 作者
    private String author;
}
